aws/resource_aws_cloudsearch_domain.go

```diff
@@ -0,0 +1,852 @@
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"service_access_policies": {
+				Type:             schema.TypeString,
+				ValidateFunc:     validateIAMPolicyJson,
+				Required:         true,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+			},
+
+			"index": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateIndexName,
+						},
+
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateIndexType,
+						},
+
+						"search": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"facet": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"return": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"sort": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"highlight": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"analysis_scheme": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"default_value": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"wait_for_endpoints": {
+				Type:             schema.TypeBool,
+				Optional:         true,
+				Default:          true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool { return true },
+			},
+
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"document_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"search_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+// Terraform CRUD Functions
+func resourceAwsCloudSearchDomainCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudsearchconn
+
+	input := cloudsearch.CreateDomainInput{
+		DomainName: aws.String(d.Get("name").(string)),
+	}
+
+	output, err := conn.CreateDomain(&input)
+	if err != nil {
+		log.Printf("[DEBUG] Creating CloudSearch Domain: %#v", input)
+		return fmt.Errorf("%s %q", err, d.Get("name").(string))
+	}
+
+	d.SetId(aws.StringValue(output.DomainStatus.ARN))
+	return resourceAwsCloudSearchDomainUpdate(d, meta)
+}
+
+func resourceAwsCloudSearchDomainRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudsearchconn
+
+	domainlist := cloudsearch.DescribeDomainsInput{
+		DomainNames: []*string{
+			aws.String(d.Get("name").(string)),
+		},
+	}
+
+	resp, err := conn.DescribeDomains(&domainlist)
+	if err != nil {
+		return err
+	}
+	domain := resp.DomainStatusList[0]
```

Comment:
this is throwing an index out-of-range panic error here when running `plan`. Any ideas to the cause would be super helpful! I'm using one of the sample configs from the acceptance tests for some additional context.

`panic: runtime error: index out of range [0] with length 0`

Reply:
I'd check the name of your domain and ensure that it matches the one you're expecting in your AWS account. From this it just seems like it wasn't able to find the domain from the given name where it is expecting one to be. Code could definitely be improved upon here with some additional sanity checks.